pkg/build/api/v1beta3/conversion.go

```diff
@@ -0,0 +1,210 @@
+				return err
+			}
+			return nil
+		},
+		func(in *newer.BuildParameters, out *BuildConfigSpec, s conversion.Scope) error {
+			err := s.DefaultConvert(&in.Strategy, &out.Strategy, conversion.IgnoreMissingFields)
+			if err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Source, &out.Source, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Output, &out.Output, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Revision, &out.Revision, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Resources, &out.Resources, 0); err != nil {
+				return err
+			}
+			return nil
+		},
+		func(in *BuildConfigSpec, out *newer.BuildParameters, s conversion.Scope) error {
+			err := s.DefaultConvert(&in.Strategy, &out.Strategy, conversion.IgnoreMissingFields)
+			if err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Source, &out.Source, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Output, &out.Output, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Revision, &out.Revision, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Resources, &out.Resources, 0); err != nil {
+				return err
+			}
+			return nil
+		}, func(in *newer.STIBuildStrategy, out *STIBuildStrategy, s conversion.Scope) error {
+			return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
+		},
+		func(in *STIBuildStrategy, out *newer.STIBuildStrategy, s conversion.Scope) error {
+			return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
+		},
+		// Rename DockerBuildStrategy.BaseImage to DockerBuildStrategy.Image
+		func(in *newer.DockerBuildStrategy, out *DockerBuildStrategy, s conversion.Scope) error {
+			return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
+		},
+		func(in *DockerBuildStrategy, out *newer.DockerBuildStrategy, s conversion.Scope) error {
+			return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
+		},
+		func(in *newer.CustomBuildStrategy, out *CustomBuildStrategy, s conversion.Scope) error {
+			return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
+		},
+		func(in *CustomBuildStrategy, out *newer.CustomBuildStrategy, s conversion.Scope) error {
+			return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
+		},
+		// Deprecate ImageTag and Registry, replace with To / Tag / DockerImageReference
+		func(in *newer.BuildOutput, out *BuildOutput, s conversion.Scope) error {
+			return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
+		},
+		func(in *BuildOutput, out *newer.BuildOutput, s conversion.Scope) error {
+			return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
+		},
+		// Rename ImageRepositoryRef to From
+		func(in *newer.ImageChangeTrigger, out *ImageChangeTrigger, s conversion.Scope) error {
+			// Note we lose the From/Image data in the ICT here if it was present,
+			// that data no longer has any value.
```

Comment:
this isn't where we lose that data.  we lost it during v1beta1->internal conversion, not here where both internal and v1beta3 don't have that info.